store/sqlstore/post_store.go

```diff
@@ -747,47 +747,46 @@ var specialSearchChar = []string{
 	":",
 }
 
-func (s *SqlPostStore) Search(teamId string, userId string, params *model.SearchParams) store.StoreChannel {
-	return store.Do(func(result *store.StoreResult) {
-		queryParams := map[string]interface{}{
-			"TeamId": teamId,
-			"UserId": userId,
-		}
+func (s *SqlPostStore) Search(teamId string, userId string, params *model.SearchParams) (*model.PostList, *model.AppError) {
 
-		termMap := map[string]bool{}
-		terms := params.Terms
+	queryParams := map[string]interface{}{
+		"TeamId": teamId,
+		"UserId": userId,
+	}
 
-		if terms == "" && len(params.InChannels) == 0 && len(params.FromUsers) == 0 && len(params.OnDate) == 0 && len(params.AfterDate) == 0 && len(params.BeforeDate) == 0 {
-			result.Data = []*model.Post{}
-			return
-		}
+	termMap := map[string]bool{}
+	terms := params.Terms
 
-		searchType := "Message"
-		if params.IsHashtag {
-			searchType = "Hashtags"
-			for _, term := range strings.Split(terms, " ") {
-				termMap[strings.ToUpper(term)] = true
-			}
-		}
+	if terms == "" && len(params.InChannels) == 0 && len(params.FromUsers) == 0 && len(params.OnDate) == 0 && len(params.AfterDate) == 0 && len(params.BeforeDate) == 0 {
+		return nil, model.NewAppError("SqlPostStore.Search", "store.sql_post.search.app_error", nil, "userId="+userId+" teamId="+teamId, http.StatusBadRequest)
+	}
 
-		// these chars have special meaning and can be treated as spaces
-		for _, c := range specialSearchChar {
-			terms = strings.Replace(terms, c, " ", -1)
+	searchType := "Message"
+	if params.IsHashtag {
+		searchType = "Hashtags"
+		for _, term := range strings.Split(terms, " ") {
+			termMap[strings.ToUpper(term)] = true
 		}
+	}
 
-		var posts []*model.Post
+	// these chars have special meaning and can be treated as spaces
+	for _, c := range specialSearchChar {
+		terms = strings.Replace(terms, c, " ", -1)
+	}
 
-		deletedQueryPart := "AND DeleteAt = 0"
-		if params.IncludeDeletedChannels {
-			deletedQueryPart = ""
-		}
+	var posts []*model.Post
 
-		userIdPart := "AND UserId = :UserId"
-		if params.SearchWithoutUserId {
-			userIdPart = ""
-		}
+	deletedQueryPart := "AND DeleteAt = 0"
+	if params.IncludeDeletedChannels {
+		deletedQueryPart = ""
+	}
+
+	userIdPart := "AND UserId = :UserId"
+	if params.SearchWithoutUserId {
+		userIdPart = ""
+	}
 
-		searchQuery := `
+	searchQuery := `
 			SELECT
```

Comment:
Do you mind to remove one tab from each line of the string to match the previous expected indentation?